src/Board.coffee

```diff
@@ -141,46 +155,43 @@ class Board
     false
   
   clicked: (mouseEvent) ->
-    # TODO - [scollins] click event handler has a bug, top left corner of
-    # board does not register click event, first clickable square in top left
-    # corner redraws the entire board down and to the right, "glitchily"
-    # however, the board is "less glitchy" at a window size of 891x427
-    mouseX = mouseEvent.clientX or mouseEvent.x
-    mouseY = mouseEvent.clientY or mouseEvent.y
-    clientRect = @game.canvas.getBoundingClientRect()
-    {x, y} = @getTransformedMouseCoordinates mouseX - clientRect.left, mouseY - clientRect.top
-    return if @invalidClick x, y
-    
-    column = x / TILE_WIDTH | 0
-    row = y / TILE_HEIGHT | 0
-    tile = @tileAt column, row
-    
-    if tile
-      if not tile.revealed
-        # flip the tile
-        tile.reveal()
-        # tell the game what was revealed by the click
-        payload =
-          mouseX: mouseX
-          mouseY: mouseY
-          row: row
-          column: column
-          tile: tile.kind
-        message =
-          event: 'revealed-tile'
-          payload: payload
-        @game.sendMessage message, @, @game
-      else
-        # special case for clicking twice on a coin, we collect the coin on second click
-        if tile.kind is 'coin'
-          tile.kind = 'dirt'
+    if not @pauseCollectCoins
```

Comment:
this probably works like a charm, however handling such logic would be better inside the Game code by providing a function that handles the logic and then calls the clicked method of the board if the input is enabled

the reason being that there is quite a bit of nesting going on in this method and it's beginning to get cluttered and hard to decipher